test/test_broker.hpp

```diff
@@ -703,6 +703,38 @@ class test_broker {
         sessions_.clear();
     }
 
+    void clear_all_retained_topics() {
+        retains_.clear();
+    }
+
+    template <typename T>
+    static MQTT_NS::optional<T> get_property(MQTT_NS::v5::properties const &props) {
+        MQTT_NS::optional<T> result;
+
+        auto visitor = MQTT_NS::make_lambda_visitor(
+            [&result](T const& t) { result = t; },
+            [](auto&& ...) { }
+        );
+
+        for (auto const& p : props) {
+            MQTT_NS::visit(visitor, p);
+        }
+
+        return result;
+    }
+
+    template <typename T>
+    static void set_property(MQTT_NS::v5::properties const &props, T&& v) {
+        auto visitor = MQTT_NS::make_lambda_visitor(
+            [v = std::forward<T>(v)](T& t) mutable { t = std::forward<T>(v); },
```

Comment:
```suggestion
            [&v](T& t) mutable { t = std::forward<T>(v); },
```

It is a little bit complicated.
1. visit() is sync function. So you don't need to care about the lifetime of v.
2. The left hand side of `v` in the capture list is l-value. So the lambda body should be `{ t = MQTT_NS::force_move(v);` because captured `v` is l-value.
3. But you don't need do that. Simply capture v as reference then use std::forward in the body of lambda expression.

See https://wandbox.org/permlink/PupJ8O8Yx0ulxbjF